js/style/style_batch.js

```diff
@@ -46,11 +48,22 @@ styleBatch.prototype = {
         if (this._style._layers[layer.id] !== undefined) {
             throw new Error('There is already a layer with this ID');
         }
+
+
         if (!(layer instanceof StyleLayer)) {
+            if (validateStyle.emitErrors(this._style, validateStyle.layer({
+                value: layer,
+                style: this._style.serialize(),
+                styleSpec: styleSpec
+            }))) return this;
```

Comment:
We have a couple options here:

1) We could save a snapshot of the style (yay `getStyle`) before the batch applies and "roll back" if we encounter any validation errors
2) We could try to validate inputs as `StyleBatch` inputs are queued up, rather than 
3) We could ignore this problem. By the time a validation error occurs, most applications will be in too a broken state to continue operating normally, even if we revert. I don't think developers generally expect transaction rollbacks in APIs like this. 

Thoughts @scothis?